src/components/StudyRoom/StudyRoomSettings.js

```diff
@@ -0,0 +1,203 @@
+    //const [errorMessage, setErrorMessage] = useState('')
+    const[loading,setLoading] = useState(true);
+    const [roomData, setRoomData] = React.useState({
+        owner:'',
+        sid:'',
+        title:'',
+        color: '',
+        avatarText:'',
+        description:'',
+        participants:[],
+    });
+
+    const fetchData = useCallback(( ) => {
+        setRoomData({...roomData,
+            owner: studyRoom.owner,
+            sid: studyRoom.sid,
+            title: studyRoom.title,
+            color: studyRoom.color,
+            avatarText: studyRoom.avatarText,
+            participants: studyRoom.participants,
+            createdOn: studyRoom.createdOn,
+            description: studyRoom.description
+        })
+        setLoading(false);
+        //let email = JSON.parse(localStorage.getItem("email"));
+        // axios.get(`http://localhost:5000/room/${email}`)
+        //     .then(res => {
+        //         const data = res.data[0];
+        //         console.log(data);
+        //         setRoomData({...roomData, title: data.title,avatarText: data.avatarText,color: data.color,description: data.description});
+        //         console.log(roomData)
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+     },[])
+
+    React.useEffect(()=>{
+        fetchData();
+        let email = JSON.parse(localStorage.getItem("email"));
+        //user needs to be logged in to access
+        if(email === undefined || email === ''){
+            navigate("/login");
+        }
+        // only an owner can access the settings page
+        if(roomData.owner !== email && roomData.owner != ''){
+            navigate("/study-room-home");
+        }
+
+    },[loading])
+
+    // sends updated field to db
+    function handleUpdate(e){
+        e.preventDefault();
+        let avatarIconText = SetAvatarText(roomData.title);
+        setRoomData({...roomData, avatarText: avatarIconText});
+        // axios.post('http://localhost:5000/room/',roomData)
+        //     .then(res => {
+        //         console.log(res);
+        //         navigate("/study-room-home");
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //API call delete room
+    function deleteRoom(){
+        // axios.post('http://localhost:5000/room/delete',{username:roomData.owner, sID:roomData.sid})
+        //     .then(res => {
+        //         console.log(res);
+        //         navigate("/study-room-home");
+        //     })
+        //     .catch(err => {console.log(`Error: ${err}`); setErrorMessage(`${err}`.substring(44) === 401 ? 'request could not be sent' : `${err}`)});
+    }
+
+    //deletes the room in db
+    function handleDelete(e){
+        deleteRoom();
```

Comment:
you can put the content of the function deleteRoom in here, no need to call a function in a function here

Reply:
I was told it was better to put axios in a separate method and call that method in here

Reply:
in any case ill just put the axios in there